### DISCUSSION TITLE
How do I use LaTeX physics with Typst?

### DISCUSSION THREAD

**Discussion by alex180500** [quarto-dev/quarto-cli#11059] [Q&A] [answered]
### Description

Hello! I'm trying to use formulas in quarto with the typst backend that uses the physics package. For example:
```latex
$\ket{psi}$
```
Is there any way to include these with typst?

**Comment by mcanouil** [comment]
I believe you are in the wrong place.
You should look in Typst documentation to know if it supports this and how.

**Reply by alex180500** [reply]
I thought math is converted as TeX math, in fact I get this warning:
```
[WARNING] Could not convert TeX math \ket{\psi}, rendering as TeX:
  \ket{\psi}
      ^
  unexpected control sequence \ket
  expecting "%", "\\label", "\\tag", "\\nonumber", whitespace or "\\allowbreak"
```

**Reply by mcanouil** [reply]
That would be Pandoc's writer.

You can try using pure Typst to make your equation work if it is supported similarly to https://github.com/jgm/pandoc/issues/9294

Then look for already opened issues (or discussions): https://github.com/jgm/pandoc/issues

But before reporting anything on Pandoc, please do ensure it's supported by Typst in the first place.

edit: I don't think Typst supports this (https://typst.app/docs/reference/math/)

```sh
echo '$ ket(psi) $' > demo.typ
quarto typst compile demo.typ
```

**Reply by mcanouil** [reply]
> I thought math is converted as TeX math, in fact I get this warning:
> 
> ```
> [WARNING] Could not convert TeX math \ket{\psi}, rendering as TeX:
>   \ket{\psi}
>       ^
>   unexpected control sequence \ket
>   expecting "%", "\\label", "\\tag", "\\nonumber", whitespace or "\\allowbreak"
> ```

That does not seem to be a Typst error. (maybe that's Pandoc)
Please provide a fully reproducible example.

**Reply by alex180500** [reply]
Yeah typst does not support ket yet but there is the physica package: https://typst.app/universe/package/physica
This has support for all physics functions, but I do not think there is support for typst packages yet in quarto

**Reply by mcanouil** [reply, marked as answer]
Support for Typst packages (as LaTeX packages, there is an auto-install though) does not come from Quarto.

Use `include-*` and partials/templates to add them:

- https://quarto.org/docs/output-formats/typst.html#includes
- https://quarto.org/docs/journals/templates.html#typst-partials
